NullPointerException executing indexed delete statement

### Describe the bug

The issue happened during the nightly test. The test was restarted this morning and passed, suggesting the problem is random. It happened during the cleanup where the test realm was removed.

* 5 caches are indexed
* Same query format: `DELETE FROM <entity> WHERE realmId IN (<realmid>)`. The `<realmid>` is just a single realm.
* The field `realmId` is indexed with `@Basic` in all caches.
* Virtual Threads are enabled.

Suspects/Theory
This line https://github.com/infinispan/infinispan/blob/dbfffb0c1f3ec409544ba0529d159b15377c28e0/query/src/main/java/org/infinispan/query/clustered/commandworkers/CQDelete.java#L39
A concurrent removal with the delete statement causes `cache::removeAsync` to return null. 

The exception is the only thing logged and the cache name is unknown since it is not part of the exception. 
```
07:56:41,239 ERROR () [org.infinispan.server.hotrod.BaseRequestProcessor] ISPN005003: Exception reported org.hibernate.search.util.common.SearchException: Exception while searching locally
	at org.infinispan.query.clustered.ClusteredQueryInvoker.broadcast(ClusteredQueryInvoker.java:105)
	at org.infinispan.query.clustered.DistributedIndexedQueryImpl.executeStatement(DistributedIndexedQueryImpl.java:191)
	at org.infinispan.query.clustered.DistributedIndexedQueryImpl.execute(DistributedIndexedQueryImpl.java:154)
	at org.infinispan.query.dsl.embedded.impl.EmbeddedLuceneQuery.execute(EmbeddedLuceneQuery.java:85)
	at org.infinispan.query.core.impl.DelegatingQuery.execute(DelegatingQuery.java:111)
	at org.infinispan.query.remote.impl.BaseRemoteQueryManager.executeQuery(BaseRemoteQueryManager.java:63)
	at org.infinispan.query.remote.impl.QueryFacadeImpl.query(QueryFacadeImpl.java:45)
	at org.infinispan.server.hotrod.HotRodServer.query(HotRodServer.java:150)
	at org.infinispan.server.hotrod.CacheRequestProcessor.queryInternal(CacheRequestProcessor.java:542)
	at org.infinispan.server.hotrod.CacheRequestProcessor.lambda$query$21(CacheRequestProcessor.java:537)
	at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(ThreadPerTaskExecutor.java:314)
	at java.base/java.lang.VirtualThread.run(VirtualThread.java:329)
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException: The CompletionStage terminated with null.
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at org.infinispan.commons.util.concurrent.CompletableFutures.await(CompletableFutures.java:124)
	at org.infinispan.query.clustered.ClusteredQueryInvoker.broadcast(ClusteredQueryInvoker.java:88)
	... 11 more
Caused by: java.lang.NullPointerException: The CompletionStage terminated with null.
	at io.reactivex.rxjava3.internal.jdk8.FlowableFromCompletionStage$CompletionStageHandler.accept(FlowableFromCompletionStage.java:71)
	at io.reactivex.rxjava3.internal.jdk8.FlowableFromCompletionStage$CompletionStageHandler.accept(FlowableFromCompletionStage.java:50)
	at io.reactivex.rxjava3.internal.jdk8.FlowableFromCompletionStage$BiConsumerAtomicReference.accept(FlowableFromCompletionStage.java:91)
	at io.reactivex.rxjava3.internal.jdk8.FlowableFromCompletionStage$BiConsumerAtomicReference.accept(FlowableFromCompletionStage.java:82)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.invokeQueuedHandlers(QueueAsyncInvocationStage.java:112)
	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.apply(QueueAsyncInvocationStage.java:94)
	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.invokeQueuedHandlers(QueueAsyncInvocationStage.java:124)
	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.apply(QueueAsyncInvocationStage.java:94)
	at org.infinispan.interceptors.impl.SimpleAsyncInvocationStage.addCallback(SimpleAsyncInvocationStage.java:68)
	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.invokeQueuedHandlers(QueueAsyncInvocationStage.java:132)
	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.accept(QueueAsyncInvocationStage.java:87)
	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.accept(QueueAsyncInvocationStage.java:33)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at org.infinispan.remoting.transport.AbstractRequest.complete(AbstractRequest.java:69)
	at org.infinispan.remoting.transport.impl.MultiTargetRequest.onResponse(MultiTargetRequest.java:111)
	at org.infinispan.remoting.transport.impl.RequestRepository.addResponse(RequestRepository.java:51)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processResponse(JGroupsTransport.java:1556)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processMessage(JGroupsTransport.java:1457)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport$ChannelCallbacks.up(JGroupsTransport.java:1649)
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/query/src/main/java/org/infinispan/query/clustered/commandworkers/CQDelete.java
+++ b/query/src/main/java/org/infinispan/query/clustered/commandworkers/CQDelete.java
@@ -4,6 +4,7 @@ import static org.infinispan.query.core.impl.Log.CONTAINER;
 
 import java.util.BitSet;
 import java.util.List;
+import java.util.Objects;
 import java.util.concurrent.CompletionStage;
 import java.util.stream.Collectors;
 
@@ -36,8 +37,9 @@ final class CQDelete extends CQWorker {
       SearchQueryBuilder query = queryDefinition.getSearchQueryBuilder();
       return blockingManager.supplyBlocking(() -> fetchIds(query), this)
             .thenCompose(queryResult -> CompletionStages.performConcurrently(queryResult, concurrencyLevel,
-                  Schedulers.from(new WithinThreadExecutor()), cache::removeAsync,
-                  Collectors.summingInt(prev -> prev != null ? 1 : 0)))
+                  Schedulers.from(new WithinThreadExecutor()), (key) -> cache.removeAsync(key)
+                        .thenApply(Objects::nonNull),
+                  Collectors.summingInt(prev -> prev ? 1 : 0)))
             .thenApply(QueryResponse::new);
    }
 

